Lib/test/test_venv.py

```diff
@@ -791,6 +791,57 @@ def test_venv_same_path(self):
                 else:
                     self.assertFalse(same_path(path1, path2))
 
+    @requires_subprocess()
+    @unittest.skipIf(os.name == 'nt', 'not relevant on Windows')
+    @unittest.skipUnless(can_symlink(), 'Needs symlinks')
+    def test_executable_symlink(self):
+        """
+        Test creation using a symlink to python executable.
+        """
+        rmtree(self.env_dir)
+        with tempfile.TemporaryDirectory() as symlink_dir:
+            executable_symlink = os.path.join(
+                os.path.realpath(symlink_dir),
+                os.path.basename(sys.executable))
+            os.symlink(os.path.abspath(sys.executable), executable_symlink)
+            cmd = [executable_symlink, "-m", "venv", "--without-pip",
+                   self.env_dir]
+            subprocess.check_call(cmd)
+        data = self.get_text_file_contents('pyvenv.cfg')
+        executable = sys._base_executable
+        path = os.path.dirname(executable)
+        self.assertIn('home = %s' % path, data)
+        self.assertIn('executable = %s' %
+                      os.path.realpath(sys.executable), data)
+
+    @requires_subprocess()
+    @unittest.skipIf(os.name == 'nt', 'not relevant on Windows')
+    @unittest.skipUnless(can_symlink(), 'Needs symlinks')
+    @requireVenvCreate
+    def test_tree_symlink(self):
+        """
+        Test creation using a symlink to python tree.
+        """
+        rmtree(self.env_dir)
+        executable_abspath = os.path.abspath(sys._base_executable)
+        tree_abspath = os.path.dirname(os.path.dirname(executable_abspath))
+        with tempfile.TemporaryDirectory() as symlink_dir:
+            tree_symlink = os.path.join(
+                os.path.realpath(symlink_dir),
+                os.path.basename(tree_abspath))
+            executable_symlink = os.path.join(
+                tree_symlink,
+                os.path.basename(os.path.dirname(executable_abspath)),
+                os.path.basename(sys._base_executable))
+            os.symlink(tree_abspath, tree_symlink)
+            cmd = [executable_symlink, "-m", "venv", "--without-pip",
+                   self.env_dir]
+            subprocess.check_call(cmd)
+        data = self.get_text_file_contents('pyvenv.cfg')
+        self.assertIn('home = %s' % tree_symlink, data)
+        self.assertIn('executable = %s' %
```

Comment:
nitpick: Since this is so path-operation heavy, it would be nice to use `pathlib` here instead.

Reply:
done